src/main/java/graphql/validation/rules/DeferDirectiveLabel.java

```diff
@@ -0,0 +1,41 @@
+package graphql.validation.rules;
+
+import graphql.Internal;
+import graphql.language.Argument;
+import graphql.language.Directive;
+import graphql.language.Node;
+import graphql.language.StringValue;
+import graphql.validation.AbstractRule;
+import graphql.validation.ValidationContext;
+import graphql.validation.ValidationErrorCollector;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import static graphql.validation.ValidationErrorType.DuplicateArgumentNames;
+
+@Internal
+public class DeferDirectiveLabel extends AbstractRule {
+    private Set<String> labels = new LinkedHashSet<>();
+    public DeferDirectiveLabel(ValidationContext validationContext, ValidationErrorCollector validationErrorCollector) {
+        super(validationContext, validationErrorCollector);
+    }
+
+    @Override
+    public void checkDirective(Directive directive, List<Node> ancestors) {
+        if (!directive.getName().equals("defer") || directive.getArguments().size() == 0) {
+            return;
+        }
+        Argument labelArgument = directive.getArgument("label");
+        // argument type is validated in DeferDirectiveArgumentType
+        if (labelArgument != null && labelArgument.getValue() instanceof StringValue) {
+            if (labels.contains(((StringValue) labelArgument.getValue()).getValue())) {
```

Comment:
should we check the value of `if` before flagging a label as duplicate?
like, should this be considered valid:
```
query {
  ... @defer(if: false, label: "a") {
   a
  }
  ... @defer(if: true, label: "a") {
   anotherA
  }
}
```

Reply:
if you pull out (StringValue) labelArgument.getValue() to a variable there will be less code later

Reply:
> should we check the value of `if` before flagging a label as duplicate? like, should this be considered valid:
> 
> ```
> query {
>   ... @defer(if: false, label: "a") {
>    a
>   }
>   ... @defer(if: true, label: "a") {
>    anotherA
>   }
> }
> ```

As far as I know and checked Label must be unique regardless if value
<img width="1695" alt="Screenshot 2024-02-22 at 10 46 02 pm" src="https://github.com/graphql-java/graphql-java/assets/137855981/bd0479d2-b028-4f2e-8309-eda428be1caa">